app/src/main/java/com/iic/shopingo/api/Server.java

```diff
@@ -0,0 +1,92 @@
+package com.iic.shopingo.api;
+
+import android.net.Uri;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.HashMap;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.params.BasicHttpParams;
+import org.apache.http.params.HttpParams;
+
+/**
+ * Created by assafgelber on 3/10/15.
+ */
+public class Server {
+  private static final String HEADER_AUTH_KEY = "X-SHOPINGO-FBID";
+
+  private static final Uri BASE_URI = Uri.parse("http://iic-shopingo.herokuapp.com/");
+
+  private final HttpClient client = new DefaultHttpClient();
+
+  private String authToken;
+
+  public Server() {}
+
+  public Server(String authToken) {
+    this.authToken = authToken;
+  }
+
+  public <T> T get(String path, Class<T> responseClass) throws IOException {
+    return get(path, responseClass, new HashMap<String, Object>());
+  }
+
+  public <T> T get(String path, Class<T> responseClass, HashMap<String, Object> params) throws IOException {
```

Comment:
It's better to use the base reference type for the map instead of the implementation (i.e `Map<String, Object>`), the method shouldn't care if you pass a HashMap or an ArrayMap or a Hashtable.
It only need to know that it's a Map.